cdc/log.cc

```diff
@@ -1282,19 +1257,15 @@ class transformer final : public change_processor {
                     if (r.row().deleted_at()) {
                         _touched_parts.set<stats::part_type::ROW_DELETE>();
                         cdc_op = operation::row_delete;
-                        if (pirow && pikey) {
+                        if (pikey) {
                             for (const column_definition& column: _schema->regular_columns()) {
-                                if (!pirow->has(column.name_as_text())) {
```

Comment:
Why is it ok to drop this? Won't line 1263 dereference unset optional (*value) when a column did not exist before?

Reply:
It's not OK, it's a bug :) thanks for spotting it. I'll fix it, even though this logic here is changed in later commits.

Preimage row may not exist for this ck, so `get_col_from_row_state` will return nullopt for each regular column, and we could hit this bug.

Otherwise we should get all regular columns of the deleted row, because we request them explicitly (see [here](https://github.com/piodul/scylla/blob/cdc-pre-postimage-persistence/cdc/log.cc#L1415))...

...unless there is a possibility of a schema changing in-flight and us racing with columns being added to the schema? In this case, we should also additionally check that `value` is set.